IBAnimatable/AnimatedPresenting.swift

```diff
@@ -0,0 +1,38 @@
+//
+//  Created by Tom Baranes on 16/07/16.
+//  Copyright © 2016 Jake Lin. All rights reserved.
+//
+
+import UIKit
+
+public typealias FrameTransformer = (finalFrame: CGRect, containerFrame: CGRect) -> CGRect
+
+public protocol AnimatedPresenting: UIViewControllerAnimatedTransitioning {
```

Comment:
We may be able to have an abstract protocol like `ViewControllerAnimatedTransitioning` (name can be varied) to encapsulate `transitionDuration`, `func retrieveTransitionDuration(transitionContext: UIViewControllerContextTransitioning?) -> NSTimeInterval`, `retrieveViews` and `retrieveViewControllers`, then let `AnimatedPresenting` and `AnimatedTransitioning` conform to it.